src/dos/dos_classes.cpp

```diff
@@ -308,6 +309,119 @@ bool DOS_PSP::SetNumFiles(uint16_t file_num)
 	return true;
 }
 
+static constexpr auto bytes_to_read = 1024;
+
+std::optional<std::string> DOS_PSP::GetEnvStr(const std::string_view entry) const
+{
+	/* Walk through the internal environment and see for a match */
+	PhysPt env_read = PhysicalMake(GetEnvironment(), 0);
+
+	char env_string[bytes_to_read + 1];
+	if (entry.empty()) {
+		return {};
+	}
+
+	for (;;) {
+		MEM_StrCopy(env_read, env_string, bytes_to_read);
+		if (!env_string[0]) {
+			return {};
+		}
+		env_read += (PhysPt)(safe_strlen(env_string) + 1);
+		char* equal = strchr(env_string, '=');
+		if (!equal) {
+			continue;
+		}
+		/* replace the = with \0 to get the length */
+		*equal = '\0';
+		if (strlen(env_string) != entry.size()) {
+			continue;
+		}
+		if (!iequals(entry, env_string)) {
+			continue;
+		}
+
+		return env_string + entry.size() + sizeof('=');
+	}
+}
+
+std::vector<std::string> DOS_PSP::GetAllEnvVars() const
+{
+	auto all_env_vars = std::vector<std::string>{};
+
+	char env_string[bytes_to_read + 1];
+	PhysPt env_read = PhysicalMake(GetEnvironment(), 0);
+	for (;;) {
+		MEM_StrCopy(env_read, env_string, bytes_to_read);
+		if (!env_string[0]) {
+			return all_env_vars;
+		}
+		all_env_vars.emplace_back(env_string);
+		env_read += (PhysPt)(safe_strlen(env_string) + 1);
+	}
+}
+
+bool DOS_PSP::SetEnv(std::string_view entry, std::string_view new_string)
```

Comment:
So following the above line of thought, `entry`, should become `var_name` and `new_string` will be `value`.